src/config/options.rs

```diff
@@ -49,6 +49,19 @@ pub enum ControlBraceStyle {
     AlwaysNextLine,
 }
 
+#[config_type]
+/// How to treat the semicolon that is optional for final diverging expressions
+/// (`return`/`break`/`continue`).
+pub enum TrailingSemicolon {
+    /// Always rewrite `return;` and `break;` expressions to have a trailing semicolon,
+    /// unless the block is a single-line block, e.g. `let PAT = e else { return }`.
```

Comment:
This is kind of weird, but preexisting behavior, I believe.

I don't exactly understand what rustfmt does to avoid adding a semicolon for single-line-style blocks.

Reply:
There are different kinds of blocks. What kinds of blocks are you referencing? Just `let-else` blocks?

To the best of my knowledge all `let-else` blocks are handled when rewriting `ast::Local` nodes. The code removes 1 from the shape to account for the trailing semicolon (line 127). Maybe it shouldn't?

https://github.com/rust-lang/rustfmt/blob/4b1596f8a12f669990721efe5ba4b31a75b75c6f/src/items.rs#L49-L188

Reply:
I'm pretty sure all block rewriting eventually calls into [`rewrite_block_inner`](https://github.com/rust-lang/rustfmt/blob/4b1596f8a12f669990721efe5ba4b31a75b75c6f/src/expr.rs#L587-L615), which eventually delegates to a few other block rewrite functions, e.g `rewrite_empty_block` or `rewrite_single_line_block`, but I'm fairly certain none of the block rewriting code handles trailing semicolons.